dialog.cpp

```diff
@@ -201,9 +204,43 @@ void Dialog::moveEvent(QMoveEvent *event)
  ************************************************/
 void Dialog::showEvent(QShowEvent *event)
 {
-    connect(KX11Extras::self(), &KX11Extras::activeWindowChanged, this, &Dialog::onActiveWindowChanged);
-    connect(KX11Extras::self(), &KX11Extras::currentDesktopChanged, this, &Dialog::onCurrentDesktopChanged);
-    return QDialog::showEvent(event);
+    if (QGuiApplication::platformName() == QSL("wayland"))
+    {
+        winId();
+        if (QWindow *win = windowHandle())
+        {
+            if (LayerShellQt::Window* layershell = LayerShellQt::Window::get(win))
+            {
+                layershell->setLayer(LayerShellQt::Window::Layer::LayerTop);
+                layershell->setKeyboardInteractivity(LayerShellQt::Window::KeyboardInteractivityOnDemand);
```

Comment:
Most probably we need to set this to `KeyboardInteractivityExclusive` as in sway and latest labwc it only gets focus by mouse action.

Reply:
The problem should be fixed in labwc (although its latest stable version has no problem), and "on-demand" should work: the window is shown and demands focus; there's noting "exclusive" about it.

Reply:
To me "exclusive" seems more logic, but if we stick with `OnDemand` we need to open a bug report in sway (and maybe others).

Reply:
If "exclusive" isn't just a word (it seems to be until now), I don't think we should use it here. Runner doesn't get the focus exclusively; it just get the focus when shown.

Reply:
See https://wayland.app/protocols/wlr-layer-shell-unstable-v1#zwlr_layer_surface_v1:enum:keyboard_interactivity:entry:exclusive. I don't like this part:

"In this case, no guarantee is made when this surface will receive keyboard focus (if ever)."

In contrast, read the explanation of `on_demand` below it. It sounds quite reasonable for us.

Reply:
That sounds convincing to leave it as it is. Opening a bug at sway when this is merged.